packages/pwa-kit-extension-sdk/src/react/classes/ApplicationExtension.ts

```diff
@@ -100,16 +100,24 @@ export class ApplicationExtension<
      * Called on the server to serialize the extension data that will be sent to the client.
      *
      * @returns SerializedExtension - The serialized extension data.
+     * @returns null if there's no asynchronous state that needs to be serialized
      * @throws Error if the routes have not been loaded.
      */
-    public serialize(): SerializedExtension {
+    public serialize(): SerializedExtension | null {
+        if (typeof this.getRoutesAsync === 'undefined') return null
```

Comment:
`serialize` (and `deserialize` too) is itself responsible for knowing when to serialize or not.